electronics_lib/Passives.py

```diff
@@ -314,6 +297,195 @@ def derated_capacitance(row: Dict[str, Any]) -> Tuple[float, float]:
+      name='Capacitor_SMD:C_0603_1608Metric',
+      max=1.1e-6,
+      derate=0,
+      vc_pairs={             50:   1e-7, 25: 1e-6, 16: 2.2e-6, 10:   1e-5},
+    ),
+    SmtCeramicCapacitorGenericPackageSpecs(
+      name='Capacitor_SMD:C_0805_2012Metric',
+      max=11e-6,
+      derate=0.08,
+      vc_pairs={100:   1e-7, 50:   1e-7, 25: 1e-5, },
+    ),
+    SmtCeramicCapacitorGenericPackageSpecs(
+      name='Capacitor_SMD:C_1206_3216Metric',
+      max=22e-6,
+      derate=0.04,
+      vc_pairs={100:   1e-7, 50: 4.7e-6, 25: 1e-5,             10: 2.2e-5},
+    ),
+    SmtCeramicCapacitorGenericPackageSpecs(
+      name='Capacitor_SMD:C_1210_3225Metric',
+      max=4.7e-5,
+      derate=0,
+      vc_pairs={100: 4.7e-6, 50:   1e-5,           16: 2.2e-5, 10: 4.7e-5},
+    ),
+    SmtCeramicCapacitorGenericPackageSpecs(
+      name='Capacitor_SMD:C_1812_4532Metric',
+      max=1e-4,
+      derate=0,
+      vc_pairs={100: 2.2e-6, 50:   1e-6, 25: 1e-5, },
+    ),
+  ]
+
+  def select_capacitor_no_prod_table(self, capacitance: RangeVal, voltage: RangeVal,
+                                     single_nominal_capacitance: RangeVal,
+                                     footprint_spec: str, derating_coeff: float) -> None:
+    """
+    Selects a generic capacitor without using product tables
+
+    :param capacitance: user-specified (derated) capacitance
+    :param voltage: user-specified voltage
+    :param single_nominal_capacitance: used when no single cap with requested capacitance, must generate multiple parallel caps,
+                                       actually refers to max capacitance for a given part
+    :param footprint_spec: user-specified package footprint
+    :param derating_coeff: user-specified derating coefficient, if used then footprint_spec must be specified
+    """
+
+    def select_package(min_nominal_capacitance: RangeVal, voltage: RangeVal) -> Optional[str]:
+
+      if footprint_spec == "":
+        package_options = self.PACKAGE_SPECS
+      else:
+        package_options = [spec for spec in self.PACKAGE_SPECS if spec.name == footprint_spec]
+
+      for package in package_options:
+        if package.max >= min_nominal_capacitance[1]:
+            for package_max_voltage, package_max_capacitance in package.vc_pairs.items():
+              if package_max_voltage >= voltage[1] and package_max_capacitance >= min_nominal_capacitance[1]:
+                return package.name
+      return None
+
+    min_nominal_capacitance = (capacitance[0] / derating_coeff, capacitance[1] / derating_coeff)
+
+    if min_nominal_capacitance[1] > single_nominal_capacitance[1]:
+      num_caps = math.ceil(min_nominal_capacitance[0] / self.SINGLE_CAP_MAX)
+      assert num_caps * self.SINGLE_CAP_MAX < min_nominal_capacitance[1], "can't generate parallel caps within max capacitance limit"
+
+      self.assign(self.selected_derated_capacitance, (
+        num_caps * capacitance[0],
+        num_caps * capacitance[1],
+      ))
```

Comment:
Are you sure about this? I'm not sure a meaningful number comes out if you multiply the original user-specified capacitance by the number of parallel capacitors?